### ISSUE TITLE
Feature request: trim reads exceeding a given length

### ISSUE THREAD

**Issue by fplaza** [BenLangmead/bowtie2#191]
Hello,

I am using bowtie2 to map Ion Torrent reads of variable length.

Current trimming options remove a fixed number of nucleotides.
Could you add an option to trim reads exceeding a given length?

Thank you.

**Comment by ch4rr0** [comment]
We have added a new command line option `--trim-reads-exceeding-len <pos:n>` which should do just that.
Where:
`pos`: can be either `3` or `5` indicating the end where reads should get trimmed 
`n`: a positive integer

You can test the command with this [build](https://github.com/BenLangmead/bowtie2/releases/tag/v2.3.4.2-alpha.2) by issuing the following command:
```
bowtie2 -x example/index/lambda_virus --trim-reads-exceeding-len 3:40 example/reads/longreads.fq
```

This command should cause `bowtie2` to trim from the `3'-end` all reads with a length greater than 40 bases. We are currently deciding whether to make this option mutually exclusive with the other two trimming options, `--trim5` and `--trim3`, so this current implementation is by no means final.

Feedback is appreciated.

**Comment by fplaza** [comment]
Thank you for implementing this feature.
The alpha build works as expected on my computer.

**Comment by ch4rr0** [comment]
We made a few changes to this command line option as detailed below. The changes have been included as part of our most recent release, v2.3.4.2

> New option `--trim-to N` causes `bowtie2` to trim reads longer
than `N` bases to exactly `N` bases from the 3' end by default. 
Users can trim from either 3'  or 5' end, e.g. `--trim-to 5:30`
trims reads to 30 bases, truncating at the 5' end.
